java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt

```diff
@@ -4101,6 +4451,163 @@ open class KotlinFileExtractor(
+                    extractExprContext(id, locId, enclosingCallable, enclosingStmt)
+                    tw.writeNamestrings(v.toString(), v.toString(), id)
+                }
+            }
+            v is Float -> {
+                exprIdOrFresh<DbFloatingpointliteral>(overrideId).also { id ->
+                    val type = useType(e.type)
+                    val locId = tw.getLocation(e)
+                    tw.writeExprs_floatingpointliteral(id, type.javaResult.id, parent, idx)
+                    tw.writeExprsKotlinType(id, type.kotlinResult.id)
+                    extractExprContext(id, locId, enclosingCallable, enclosingStmt)
+                    tw.writeNamestrings(v.toString(), v.toString(), id)
+                }
+            }
+            v is Double -> {
+                exprIdOrFresh<DbDoubleliteral>(overrideId).also { id ->
+                    val type = useType(e.type)
+                    val locId = tw.getLocation(e)
+                    tw.writeExprs_doubleliteral(id, type.javaResult.id, parent, idx)
+                    tw.writeExprsKotlinType(id, type.kotlinResult.id)
+                    extractExprContext(id, locId, enclosingCallable, enclosingStmt)
+                    tw.writeNamestrings(v.toString(), v.toString(), id)
+                }
+            }
+            v is Boolean -> {
+                exprIdOrFresh<DbBooleanliteral>(overrideId).also { id ->
+                    val type = useType(e.type)
+                    val locId = tw.getLocation(e)
+                    tw.writeExprs_booleanliteral(id, type.javaResult.id, parent, idx)
+                    tw.writeExprsKotlinType(id, type.kotlinResult.id)
+                    extractExprContext(id, locId, enclosingCallable, enclosingStmt)
+                    tw.writeNamestrings(v.toString(), v.toString(), id)
+                }
+            }
+            v is Char -> {
+                exprIdOrFresh<DbCharacterliteral>(overrideId).also { id ->
+                    val type = useType(e.type)
+                    val locId = tw.getLocation(e)
+                    tw.writeExprs_characterliteral(id, type.javaResult.id, parent, idx)
+                    tw.writeExprsKotlinType(id, type.kotlinResult.id)
+                    extractExprContext(id, locId, enclosingCallable, enclosingStmt)
+                    tw.writeNamestrings(v.toString(), v.toString(), id)
+                }
+            }
+            v is String -> {
+                exprIdOrFresh<DbStringliteral>(overrideId).also { id ->
+                    val type = useType(e.type)
+                    val locId = tw.getLocation(e)
+                    tw.writeExprs_stringliteral(id, type.javaResult.id, parent, idx)
+                    tw.writeExprsKotlinType(id, type.kotlinResult.id)
+                    extractExprContext(id, locId, enclosingCallable, enclosingStmt)
+                    tw.writeNamestrings(toQuotedLiteral(v.toString()), v.toString(), id)
```

Comment:
This change results in a lot of changes to expected files. Would it make sense to do this in a separate PR to make the diff cleaner?